sensor-board-firmware/src/vl6180x/vl6180x.c

```diff
@@ -0,0 +1,133 @@
+
+    /* Wait for device ready. */
+    do {
+        status = vl6180x_read_register(dev, VL6180X_RESULT_RANGE_STATUS);
+    } while ((status & (1 << 0)) == 0);
+
+    /* Return error code. */
+    return status >> 4;
+}
+
+void vl6180x_change_i2c_address(vl6180x_t *dev, uint8_t address)
+{
+    vl6180x_write_register(dev, VL6180X_I2C_SLAVE_DEVICE_ADDRESS, address);
+    dev->address = address;
+}
+
+/* Configuration as in application note AN4545: VL6180X basic ranging */
+void vl6180x_configure(vl6180x_t *dev)
+{
+    while (vl6180x_read_register(dev, 0x16) != 0x01) {
+    }
+    /* clear SYSTEM__FRESH_OUT_OF_RESET */
+    vl6180x_write_register(dev, 0x016, 0x00);
+
+    /* Mandatory: Private registers. */
+    vl6180x_write_register(dev, 0x0207, 0x01);
+    vl6180x_write_register(dev, 0x0208, 0x01);
+    // vl6180x_write_register(dev, 0x0133, 0x01); // undocumented
+    vl6180x_write_register(dev, 0x0096, 0x00);
+    vl6180x_write_register(dev, 0x0097, 0xfd);
+    vl6180x_write_register(dev, 0x00e3, 0x00);
+    vl6180x_write_register(dev, 0x00e4, 0x04);
+    vl6180x_write_register(dev, 0x00e5, 0x02);
+    vl6180x_write_register(dev, 0x00e6, 0x01);
+    vl6180x_write_register(dev, 0x00e7, 0x03);
+    vl6180x_write_register(dev, 0x00f5, 0x02);
+    vl6180x_write_register(dev, 0x00d9, 0x05);
+    vl6180x_write_register(dev, 0x00db, 0xce);
+    vl6180x_write_register(dev, 0x00dc, 0x03);
+    vl6180x_write_register(dev, 0x00dd, 0xf8);
+    vl6180x_write_register(dev, 0x009f, 0x00);
+    vl6180x_write_register(dev, 0x00a3, 0x3c);
+    vl6180x_write_register(dev, 0x00b7, 0x00);
+    vl6180x_write_register(dev, 0x00bb, 0x3c);
+    vl6180x_write_register(dev, 0x00b2, 0x09);
+    vl6180x_write_register(dev, 0x00ca, 0x09);
+    vl6180x_write_register(dev, 0x0198, 0x01);
+    vl6180x_write_register(dev, 0x01b0, 0x17);
+    vl6180x_write_register(dev, 0x01ad, 0x00);
+    vl6180x_write_register(dev, 0x00ff, 0x05);
+    vl6180x_write_register(dev, 0x0100, 0x05);
+    vl6180x_write_register(dev, 0x0199, 0x05);
+    // vl6180x_write_register(dev, 0x0109, 0x07); // undocumented
+    // vl6180x_write_register(dev, 0x010a, 0x30);
+    // vl6180x_write_register(dev, 0x003f, 0x46);
+    vl6180x_write_register(dev, 0x01a6, 0x1b);
+    vl6180x_write_register(dev, 0x01ac, 0x3e);
+    vl6180x_write_register(dev, 0x01a7, 0x1f);
+    // vl6180x_write_register(dev, 0x0103, 0x01); // undocumented
+    vl6180x_write_register(dev, 0x0030, 0x00);
+    // vl6180x_write_register(dev, 0x001b, 0x0a); // SYSRANGE__INTERMEASUREMENT_PERIOD
+    // vl6180x_write_register(dev, 0x003e, 0x0a); // SYSALS__INTERMEASUREMENT_PERIOD
+    // vl6180x_write_register(dev, 0x0131, 0x04); // undocumented
+    // // vl6180x_write_register(dev, 0x0011, 0x10);
+    // // vl6180x_write_register(dev, 0x0014, 0x24);
+    // // vl6180x_write_register(dev, 0x0031, 0xff);
+    // vl6180x_write_register(dev, 0x00d2, 0x01); // undocumented
+    // vl6180x_write_register(dev, 0x00f2, 0x01); // undocumented
```

Comment:
I suggest removing unused code, and do it in a separate commit if you want to keep it in history.

Reply:
Ok, will do. I think this is coming from different reference implementations.